### DISCUSSION TITLE
Custom DNS Issue

### DISCUSSION THREAD

**Discussion by thenabeelhassan** [vercel/community#2348] [Help] [answered]
### Summary

Hi All,

I am trying to add custom domain to my project. I have also added following DNS Entry in my domain NS, but I am still getting invalid configuration on project custom domain.

<img width="763" alt="image" src="https://user-images.githubusercontent.com/49884022/236690951-3ea91a83-5b45-4f5a-93cc-51a18184b50f.png">
<img width="711" alt="2023-05-07_21-46-26" src="https://user-images.githubusercontent.com/49884022/236691154-f025f504-65b0-4280-b17d-577ff33b01f5.png">

P.S: It's been more then 24hours of adding DNS Entry.

**Comment by thenabeelhassan** [comment]
### UPDATE

I have SSL Configured for my domain, and I am getting **The connection for this site is not secure** error.
<img width="918" alt="2023-05-08_8-18-29" src="https://user-images.githubusercontent.com/49884022/236726338-d4b5e84c-824a-41e7-a9b6-b040ff3551f3.png">

On trying with HTTP, I get following page:
<img width="960" alt="2023-05-08_8-21-52" src="https://user-images.githubusercontent.com/49884022/236726691-adf2da8e-0651-45bb-aaeb-5e10657f49f9.png">

As per [Click here to learn how to assign it to a project](https://vercel.com/docs/custom-domains#assigning-a-domain-to-a-git-branch), I am facing following error:
<img width="710" alt="2023-05-08_8-27-23" src="https://user-images.githubusercontent.com/49884022/236727354-2190ae4d-c179-4ca1-a3e9-c558ee7f1215.png">


P.S: Invalid Configuration message is still appearing on this domain.

**Comment by amyegan** [comment]
Hi @thenabeelhassan. It looks like you're using Cloudflare. Do you happen to have [Always Use HTTPS](https://vercel.com/guides/using-cloudflare-with-vercel#always-use-https) enabled? If so, try disabling it and let me know if that worked.

**Reply by thenabeelhassan** [reply]
@amyegan Well I have to use HTTPS as these projects will be having information that cannot be published over HTTP only.

Edit: I am using DNS only record and as per [How do I use a Cloudflare domain with Vercel?](https://vercel.com/guides/using-cloudflare-with-vercel) if the record is set to DNS only there is no need to disable HTTPS or make any changes on SSL/HTTPS.

**Reply by amyegan** [reply]
It looks like "Always Use HTTPS" is no longer a relevant setting. But your problem may be related to "Cloudflare Access":
> If you're using [Cloudflare Access](https://vercel.com/guides/using-cloudflare-with-vercel#cloudflare-access), this will force all website requests via the Access service, resulting in Vercel's domain verification checks failing. This makes Vercel unable to issue your TLS certificates properly. To avoid this, you must add the following path to a bypass policy in your Cloudflare Access settings: `http://<YOUR_DOMAIN>/.well-known/acme-challenge/*`

**Reply by smaeda-ks** [reply]
Looks like you sorted it out. I see a typo in your original screenshot `asasni` vs `asani`, was that the case? I checked your current project domains and saw you changed the subdomain to `asani-notifications` and it's working fine.

**Reply by thenabeelhassan** [reply, marked as answer]
@smaeda-ks I corrected that typo earlier but was still getting the error.

Today I deleted the domain from project and update by DNS to asani-notifications and set the TTL to 1 min, checked the DNS record from cmd via nslookup and after successful result, I added the domain again to the project and It works.

My takeaway from this issue: first create the record in DNS and after verification add the domain to the project.

**Comment by harshalone** [comment]
hi @amyegan in this tutorial, the proxy approach does not work. Any tips how can I make it work with proxy/

**Reply by smaeda-ks** [reply]
@harshalone how exactly it's not working? Could you please include more details?
